### DISCUSSION TITLE
How does Awakened POE send chat commands??

### DISCUSSION THREAD

**Discussion by Guy-BEANSPerformaceUK** [SnosMe/awakened-poe-trade#864] [Q&A] [answered]
How does Awakened POE send chat commands to the game without the chat window popping up?? 

It surprises me every time because most other tools that do chat commands have the window popping up etc.

**Comment by SnosMe** [comment, marked as answer]
Nothing special. Sending all key presses to the OS within ~5ms. Now the game consumes inputs from OS queue. If PoE handles everything between pressing Enter's within ~16ms frame, it obviously won't even render opened chat.

**Reply by SnosMe** [reply]
And you still sometimes see chat box open if this ~5ms interval happens on the edge of two rendered frames. Should be more often if you have 120+ fps screen.
